### ISSUE TITLE
Meeting Editor: Do not allow users to select TC without also selecting VM

### ISSUE THREAD

**Issue by jbraswell** [bmlt-enabled/bmlt-root-server#363]
Prerequisite: https://github.com/bmlt-enabled/bmlt-root-server/issues/355

The TC format should really only be used in conjunction with the VM format to indicate that a meeting that was formerly physical has a temporary virtual replacement. It is to let folks know that they shouldn't show up to the old meeting facility, but that there is a virtual option.

On the location tab, (per https://github.com/bmlt-enabled/bmlt-root-server/issues/355), do not allow users to select TC if they have not also selected VM.

Add some text to the UI explaining the purpose of the TC designation, and that if a meeting truly has temporarily shut down with no virtual replacement, then they should simply unpublish the meeting.

**Comment by GarrettWeinberg** [comment]
We were discussing the possibility of using the TC format to mark meetings that are closed due to let's say inclement weather or the church is closed for some other reason or another. I saw this and was curious about the implications.

**Comment by jbraswell** [comment]
This change would solidify TC as a modifier to the VM format. I’m not
comfortable with using it for any other reason, at least not in the near
term. To improve worldwide rampant data quality issues we’re going to need
to have less flexibility.

Note that this is also coupled with
https://github.com/bmlt-enabled/bmlt-root-server/issues/355, which removes
VM, HY, and TC from the formats tab and places them on the location tab.
Data entry users will no longer be aware that they’re modifying meeting
formats, though they’ll still be saved as formats on the backend for now to
preserve crouton functionality.

Eventually, when most of the world has moved on to the new version and
repaired their data, we can migrate these fields away from abusing formats
completely. That’s not a near-term thing, but this work is a short-term
requirement to get it right.

**Comment by tempsaint** [comment]
I don't think forcing the use of two codes for one purpose makes sense. As Garrett mentioned, right now users can use the TC code for any time a meetings has to cancel meetings due to a facility closure... a month of construction or similar.  Maybe coming up a better method of marking that *this* VM is normally an in-person meeting but is virtual for the time being.

**Comment by jbraswell** [comment]
Please consider that to the users doing data entry, per issue https://github.com/bmlt-enabled/bmlt-root-server/issues/355, they will no longer be aware that they are using format codes at all when they select VM, HY, and TC. We don’t even have to surface them in the UI as VM, HY, or TC. The checkbox can even say something like “This virtual meeting is a temporary replacement for a physical meeting, whose facility is currently closed.“.

I'd consider supporting general closures and holiday closures (outside of just unpublishing) a brand new feature request.

**Comment by jbraswell** [comment]
@tempsaint had an interesting idea. Rather than using TC as a modifier for VM in the UI, there could just be three different mutually exclusive boxes.

1. Virtual Meeting
2. Virtual Meeting that has temporarily replaced a face-to-face meeting
3. Hybrid Meeting

If someone selects #2, we'd persist both the VM and TC formats on the backend.

**Comment by tempsaint** [comment]
And to follow up on what @jbraswell just posted, maybe some logic could be used to verify... like if someone selects the VM-Only box, the system looks to see if there's a name of facility and street address and asks the user to verify. And vice-versa with the "VM Replacement" box... if there's no facility, it asks the user if they checked the correct box.

**Comment by tempsaint** [comment]
> If someone selects #2, we'd persist both the VM and TC formats on the backend.

Does this negatively affect the maximum of 5 codes that NAWS stores (if we're using two backend formats codes).

**Comment by jbraswell** [comment]
> > If someone selects #2, we'd persist both the VM and TC formats on the backend.
> 
> Does this negatively affect the maximum of 5 codes that NAWS stores (if we're using two backend formats codes).

Yeah, but it's no worse than what we're currently doing. I think, longer term, we move away from formats for these designators completely... and hopefully by then we can help NAWS with some tooling on their side to get the right data into their database. 🤞

**Comment by radius314** [comment]
Thoughts? 

![image](https://user-images.githubusercontent.com/1850811/102677485-ae384d00-4170-11eb-9bfb-a5fe91aec36c.png)

**Comment by GarrettWeinberg** [comment]
@radius314 what are the repercussions on the format codes with this implemented? Will TC still be available to select, or will this remove it altogether?

**Comment by radius314** [comment]
The way I was thinking of implementing this is that selection of the radio button would automatically select options from the format tab (and potentially grey them out).  Based off the conversation in the ticket, TC would still be selectable from the format view.  

We could also make it so that if you selected specific formats that it auto calculates the the radio button conversely.  So they are synched.  

The radio button selection wouldn't actually persist.  

We could certainly add Temporarily Closed as another radio button here too.

**Comment by GarrettWeinberg** [comment]
I personally like the idea of just selecting the radio buttons automatically setting the format and graying them out. I feel like if we allow it to go both ways, it may get confusing and cause inaccuracies. (just my opinion). I think the use case of the TC format that i originally brought up on this is a one-off, and TBH i could probably modify the Temp Closed plugin and create a new format to make it work for us.

**Comment by jbraswell** [comment]
It should be a radio button with the three discussed options, but I do not want to make TC independently selectable. Not sure I like the idea of displaying these format codes on the formats tab, but if they're disabled it might make sense.

**Comment by jbraswell** [comment]
> I personally like the idea of just selecting the radio buttons automatically setting the format and graying them out. I feel like if we allow it to go both ways, it may get confusing and cause inaccuracies. (just my opinion). I think the use case of the TC format that i originally brought up on this is a one-off, and TBH i could probably modify the Temp Closed plugin and create a new format to make it work for us. 

Agree with this. If we let folks select these formats on the formats tab it'll be super confusing - it must only be selectable by the new radios on the location tab.

**Comment by radius314** [comment]
I suppose they could be hidden or greyed out.

**Comment by radius314** [comment]
We would need the 4th option which says neither of these choices are
selected.  Not sure “In-person” is the right wording.

**Comment by jbraswell** [comment]
Yeah seems like hidden or grayed/disabled would work, agree.

**Comment by tempsaint** [comment]
I like the option of just removing them (hidden) from the formats tab... unless we're going to communicate that only 5 formats get sent to NAWS for their database and the greyed out ones count (and get prioritized).

**Comment by jbraswell** [comment]
I'm not sure the right terminology either. Maybe N/A for Not Applicable is the least likely to offend?

**Comment by radius314** [comment]
We should brainstorm that.

**Comment by radius314** [comment]
> I like the option of just removing them (hidden) from the formats tab... unless we're going to communicate that only 5 formats get sent to NAWS for their database and the greyed out ones count (and get prioritized).

I'm ok with this.

**Comment by tempsaint** [comment]
> We would need the 4th option which says neither of these choices are selected. Not sure “In-person” is the right wording. 

I've liked using in-person because virtual meetings are sort of "face-to-face" (at least Zoom where you actually see people's faces). I know we can caught up in semantics. I originally liked the the "old school" term "IRL  or "In Real Life" but I don't think that's widely known anymore. And video conferencing is kind of IRL anyway. 

But whatever is chosen, make sure to use the same language for it's own radio button and the description in the Virtual replacement one... right now you have "in-person" for one and "face-to-face" in the other.

**Comment by radius314** [comment]
> > We would need the 4th option which says neither of these choices are selected. Not sure “In-person” is the right wording.
> 
> I've liked using in-person because virtual meetings are sort of "face-to-face" (at least Zoom where you actually see people's faces). I know we can caught up in semantics. I originally liked the the "old school" term "IRL or "In Real Life" but I don't think that's widely known anymore. And video conferencing is kind of IRL anyway.
> 
> But whatever is chosen, make sure to use the same language for it's own radio button and the description in the Virtual replacement one... right now you have "in-person" for one and "face-to-face" in the other.

We could debate this on scrum on Sunday.

**Comment by jbraswell** [comment]
> > We would need the 4th option which says neither of these choices are selected. Not sure “In-person” is the right wording. 
> 
> 
> 
> I've liked using in-person because virtual meetings are sort of "face-to-face" (at least Zoom where you actually see people's faces). I know we can caught up in semantics. I originally liked the the "old school" term "IRL  or "In Real Life" but I don't think that's widely known anymore. And video conferencing is kind of IRL anyway. 
> 
> 
> 
> But whatever is chosen, make sure to use the same language for it's own radio button and the description in the Virtual replacement one... right now you have "in-person" for one and "face-to-face" in the other. 

I think I'd strongly advocate for a Not Applicable selection that is the default. You could call the section something like "Virtual Location Type", and have Not Applicable be the default. 

This avoids leaning into any specific terminology for in-person meetings.

**Comment by tempsaint** [comment]
>I think I'd strongly advocate for a Not Applicable selection that is the default. You could call the section something like "Virtual Location Type", and have Not Applicable be the default.
>
>This avoids leaning into any specific terminology for in-person meetings.

That makes sense. Although what do you put in the Virtual (that has temporarily replaced a _________ meeting). If we're going to put something there, why not use that same language in the other radio button? 

Sorry... I just saw the comment about debating it on scrum on Sunday. I'll stop now.

**Comment by rogersearle** [comment]
Is "Hybrid" clearly enough understood, and could it benefit from a brief definition similar to what is proposed for Virtual?

**Comment by radius314** [comment]
![image](https://user-images.githubusercontent.com/1850811/102724589-9c74b800-42de-11eb-9eac-760dc4d5474f.png)

**Comment by jbraswell** [comment]
Nice. I think I'd call it "Venue Type" instead of "Venue Format".

**Comment by radius314** [comment]
![image](https://user-images.githubusercontent.com/1850811/102726189-9684d400-42ea-11eb-9142-0764255265e9.png)

**Comment by radius314** [comment]
Disabled formats VM, TC, HY

![image](https://user-images.githubusercontent.com/1850811/103143968-34afda00-46ef-11eb-8490-fe31f6749f8a.png)

**Comment by tempsaint** [comment]
Would it make sense to group those somehow (and maybe group other location type formats (surveillance cameras, Wheelchair, No Smoking) and meeting format type formats)? Or is that something for a future discussion/release and not really tied to this issue?

**Comment by jbraswell** [comment]
> Would it make sense to group those somehow (and maybe group other location type formats (surveillance cameras, Wheelchair, No Smoking) and meeting format type formats)? Or is that something for a future discussion/release and not really tied to this issue? 

I thought we had agreed to remove the formats from the formats tab completely, rendering this moot.

Grouping of other format types might be a good idea but def not in scope of this release.

**Comment by radius314** [comment]
We left that discussion with the idea that it could be either way.  Changing this is fairly simple to do.

**Comment by jbraswell** [comment]
> We left that discussion with the idea that it could be either way.  Changing this is fairly simple to do.

I recall folks on the scrum call unanimously agreeing that we should hide/remove the virtual formats from the formats tab. It seemed like a very clear decision.

My memory could be wrong, I really thought that happened.

Either way, I don't really care - just don't want to add the additional scope that @tempsaint is suggesting 🙂.

**Comment by tempsaint** [comment]
Yeah, no. I'm cool not expanding the scope of this at this time.

**Comment by radius314** [comment]
![image](https://user-images.githubusercontent.com/1850811/103156802-a5e89f00-477a-11eb-95b5-9e75f89d0460.png)

**Comment by radius314** [comment]
I'm not opinionated either way.

**Comment by radius314** [comment]
It was not clear to me what the decision was, hence the choice of the first approach.

**Comment by radius314** [comment]
Radio button logic implemented, please review and let me know if this is correct: 

https://github.com/bmlt-enabled/bmlt-root-server/blob/relocate-things/main_server/local_server/server_admin/server_admin_javascript.js#L2999-L3007

**Comment by radius314** [comment]
Uploading GMT20201226-193414_Danny-G-No_1920x1298 (1).mp4…

**Comment by radius314** [comment]
https://user-images.githubusercontent.com/1850811/103158081-f0244d00-4787-11eb-8de9-6b433ca94bb7.mp4
